src/java.base/share/classes/java/io/PrintWriter.java

```diff
@@ -208,6 +208,13 @@ public PrintWriter(String fileName) throws FileNotFoundException {
              false);
     }
 
+    /* package private constructor specific to ProxyingConsole */
+    PrintWriter(Writer out, Object lock) {
```

Comment:
This constructor looks fine but maybe the comment should just say that it allows the lock object to be provided rather than mentioning ProxyingConsole here.